rclcpp/test/rclcpp/test_graph_listener.cpp

```diff
@@ -268,15 +261,15 @@ TEST_F(TestGraphListener, test_errors_graph_listener_add_remove_node) {
 /* Shutdown errors */
 TEST_F(TestGraphListener, test_graph_listener_shutdown_wait_fini_error_nothrow) {
   graph_listener()->start_if_not_started();
-  auto mock_wait_set_fini = mocking_utils::inject_on_return(
+  auto mock_wait_set_fini = mocking_utils::patch_and_return(
```

Comment:
By switching these to `patch_and_return` you're not finalizing any of these objects which is probably leaking memory in these tests. 

One possible option would be to call `patch` with a custom lambda that calls the `fini` function, but returns `RCL_RET_ERROR`. Would that work on centos, or is that why it's failing to begin with?

Reply:
I'll give it a shot.  I think that is going to fail as well on CentOS, but we'll see.

Reply:
> Would that work on centos, or is that why it's failing to begin with?

It'll fail. The sort of global PLT optimization CentOS seems to be doing is what's preventing `inject_on_return` to work to being with.

Reply:
These tests here and below actually passed in the example you provided. What trouble were they giving you?

Reply:
After some discussion with @clalancette, it appears these leaks may not be avoidable.

Reply:
Yeah, it failed.  See 84b5289, where I take a different approach.  There are still leaks, but it is not worse than before.